persistence-modules/hibernate-jpa/src/test/java/com/baeldung/hibernate/serializable/JPASerializableIntegrationTest.java

```diff
@@ -0,0 +1,76 @@
+package com.baeldung.hibernate.serializable;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class JPASerializableIntegrationTest {
+
+    private static EntityManager entityManager;
+
+    @Before
+    public void setUp() throws IOException {
+        EntityManagerFactory entityManagerFactory = Persistence.createEntityManagerFactory("com.baeldung.hibernate.serializable.h2_persistence_unit");
+        entityManager = entityManagerFactory.createEntityManager();
+        entityManager.getTransaction().begin();
+    }
+
+    @Test
+    public void givenUser_whenPersisted_thenUserWillBeFoundById() {
+        UserId userId = new UserId();
+        userId.setName("John");
+        userId.setLastName("Doe");
+        Email email = new Email();
+        email.setId(1);
+        email.setName("johndoe");
+        email.setDomain("gmail.com");
+        User user = new User(userId, email);
+
+        entityManager.persist(user);
+
+        User userDb = entityManager.find(User.class, userId);
+        assertEquals(userDb.getEmail().getName(), "johndoe");
```

Comment:
Swap the order of the arguments to match the method signature - assertEquals(expected, actual)